### PR TITLE
SNOW-981602: Fix missing clientconfig initialization, add test case

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-nl**
Fix for SNOW-981602 Implement Easy Logging using a config file. Add initialization for clientconfig and add test case

### CODE REVIEW

File: tests/test_unit_logger.c [hunk 1]
```diff
@@ -33,6 +33,45 @@ void test_log_str_to_level() {
...
 
+/**
+ * Test standard logging without client config
+ */
+void test_default_log_path() {
+  char LOG_PATH[MAX_PATH] = { 0 };
+  char LOG_LEVEL[64] = { 0 };
+
+  // Pass in empty log path
+  snowflake_global_init("", SF_LOG_WARN, NULL);
```
Comment (sfc-gh-ext-simba-hx) [inline tests/test_unit_logger.c:44]: I think the more general use case is `snowflake_global_init(NULL, SF_LOG_TRACE, NULL);` which is used in test case here: https://github.com/snowflakedb/libsnowflakeclient/blob/master/tests/utils/test_setup.c#L22
Please have that one covered as well.

Comment (sfc-gh-ext-simba-nl) [reply tests/test_unit_logger.c:83]: Added another test case for null

File: tests/test_unit_logger.c [hunk 2]
```diff
@@ -33,6 +33,84 @@ void test_log_str_to_level() {
     assert_int_equal(log_from_str_to_level(NULL), SF_LOG_FATAL);
 }
 
+/**
```
Comment (sfc-gh-dprzybysz) [inline tests/test_unit_logger.c:36]: we can remove this comment - function name is meaningful

File: tests/test_unit_logger.c [hunk 3]
```diff
@@ -500,7 +580,7 @@ int main(void) {
         cmocka_unit_test(test_client_config_log_no_path),
         cmocka_unit_test(test_client_config_stdout),
 #endif
-        cmocka_unit_test(test_log_creation),
+        //cmocka_unit_test(test_log_creation),
```
Comment (sfc-gh-dprzybysz) [inline tests/test_unit_logger.c:583]: why is this test ignored?

File: tests/test_unit_logger.c [hunk 4]
```diff
@@ -33,6 +33,84 @@ void test_log_str_to_level() {
...
+  assert_int_equal(access(LOG_PATH, F_OK), 0);
+  log_close();
+
+  // Cleanup
+  remove(LOG_PATH);
+  remove(LOG_LEVEL);
+  remove(log_path_dir);
+}
+
+/**
```
Comment (sfc-gh-dprzybysz) [inline tests/test_unit_logger.c:75]: we can remove this comment

**Review by sfc-gh-dprzybysz — CHANGES_REQUESTED** [review]

File: tests/test_unit_logger.c (hunk 3 above)
Comment (sfc-gh-ext-simba-nl) [reply tests/test_unit_logger.c:583]: Accident, forgot to unignore

**Review by sfc-gh-ext-simba-hx — APPROVED** [review]
LGTM

**Comment by sfc-gh-ext-simba-hx** [comment]
The test failure in test_column_fetch likely due to server side change on LOB size from 16MB to 128MB. The test case needs to be changed to be more flexible on expected result.
```
29: [ RUN      ] test_column_length_with_bundle_202408_disabled
29: [  ERROR   ] --- 0x1000000 != 0x8000000
29: [   LINE   ] --- /mnt/host/tests/test_column_fetch.c:1519: error: Failure!
```

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Review by sfc-gh-mkubik — APPROVED** [review]
